src/pages/Proposal.vue

```diff
@@ -0,0 +1,174 @@
+			<label for="proposal_address">Proposal Account Address</label>
+			<input
+				id="proposal_address"
+				required
+				v-model="formState.proposal_address"
+			/>
+			<label for="url">Proposal ID</label>
+			<input id="url" required v-model="formState.proposal_id" />
+			<label for="voting_date">Voting Date</label>
+			<a-range-picker
+				id="voting_date"
+				format="YYYY-MM-DD HH:mm:ss"
+				value-format="YYYY-MM-DD HH:mm:ss"
+				:disabled-date="disabledDate"
+				:disabled-time="disabledRangeTime"
+				showTime
+				v-model:value="formState.vote_date"
+				style="border-color: black"
+			/>
+			<label for="vote_options">Vote</label>
+			<select
+				name="vote_type"
+				id="vote_options"
+				required
+				v-model="formState.vote_type"
+			>
+				<option :value="ProposalType.ALGO">Algo Transfer</option>
+				<option :value="ProposalType.ASA">ASA Transfer</option>
+				<option :value="ProposalType.MESSAGE">Message</option>
+			</select>
+			<div
+				class="flexBox"
+				v-if="
+					formState.vote_type && formState.vote_type !== ProposalType.MESSAGE
+				"
+			>
+				<label for="from"> From</label>
+				<input id="from" v-model="formState.from" required />
+
+				<label for="recipient">Recipient</label>
+				<input id="recipient" v-model="formState.recipient" />
+
+				<label for="amount"> Amount </label>
+				<input id="amount" v-model="formState.amount" />
+			</div>
+			<div class="flexBox" v-if="formState.vote_type === ProposalType.ASA">
+				<label for="asa_id">ASA ID</label>
+				<input id="asa_id" v-model="formState.asaId" />
+			</div>
+			<div class="flexBox" v-if="formState.vote_type === ProposalType.MESSAGE">
+				<label for="message">Message</label>
+				<input id="message" v-model="formState.message" />
+			</div>
+			<button type="submit" class="submit_btn" @click="onSubmit">Submit</button>
+		</form>
+	</div>
+</template>
+
+<script lang="ts">
+import { ProposalFormState } from "@/types";
+import { defineComponent, reactive } from "vue";
+import ProposalStore from "../store/ProposalStore";
+import { ProposalType } from "../types/enum.types";
+
+export default defineComponent({
+	name: "AddProposal",
+	data() {
+		return {
+			ProposalType,
+		};
+	},
+	setup() {
+		const store = ProposalStore();
+		const formState = reactive<ProposalFormState>(store.$state);
+
+		const onSubmit = () => {
+			let success = false;
+			if (
+				!formState.vote_type ||
+				!formState.url ||
+				!formState.url_hash ||
+				!formState.vote_date ||
+				!formState.proposal_address
+			) {
+				success = false;
+			} else {
+				switch (formState.vote_type) {
+					case ProposalType.ASA:
+						return (success = formState.asaId ? true : false);
+					case ProposalType.MESSAGE:
+						return (success = formState.message ? true : false);
+					default:
+						return (success =
+							formState.from || formState.recipient || formState.amount
+								? true
+								: false);
+				}
+			}
```

Comment:
add a proper error when something is wrong. In fact https://www.antdv.com/components/form/ handles nicely error reporting.
We could also use: https://www.antdv.com/components/alert/